gossipsub_test.go

```diff
@@ -3675,3 +3677,48 @@ func TestPublishDuplicateMessage(t *testing.T) {
 		t.Fatal("Duplicate message should not return an error")
 	}
 }
+
+func BenchmarkAppendOrMergeRPC(b *testing.B) {
+	makeTestRPC := func(numMsgs int, msgSize int) RPC {
+		msgs := make([]*pb.Message, numMsgs)
+		payload := make([]byte, msgSize)
+		for i := range msgs {
+			msgs[i] = &pb.Message{
+				Data: payload,
+				From: []byte(strconv.Itoa(i)),
+			}
+		}
+		return RPC{
+			RPC: pb.RPC{
+				Publish: msgs,
```

Comment:
In your workload, do you see RPCs being split primarily due to many messages in a single RPC? I ask because we could add some optimizations if so.